### ISSUE TITLE
EnCompleteFunc does not pull suggestions on initial call

### ISSUE THREAD

**Issue by ChrisCoffey** [ensime/ensime-vim#170]
`EnCompleteFunc` looks like the intent is to block for updates from Ensime once Vim calls the second time with `0`, however this doesn't actually happen because the queue is empty & the read loop is skipped over. 

Once the suggestions arrive, they're available & are processed immediately if you trigger omni-completion a second time for the same cursor location.

**Comment by jvican** [comment]
True, thanks for pointing that out. The problem is that you can't re-trigger omni-completion a second time from the plugin, or at least I don't know how to do it.
  
The solution would be to modify `unqueue` and force the while loop to process at least one message by blocking. Could you please do it and check if it works? Thanks.

**Comment by ChrisCoffey** [comment]
Yup, that worked exactly as expected. The results look correct as far as I can tell, but are not accepted by NeoVim's omni-complete. Is there anything that looks incorrect about this?

Running `<C-x><C-o>` on a `String`:
```
{"refresh": "always", "words": [{"info": "", "menu": "[scala]", "kind": "(x$1: Any) => String", "word": "+"}, {"info": "", "menu": "[scala]", "kind": "() => String", "word": "trim"}, {"info": "", "menu": "[scala]", "kind": "(x$1: String) => Array[String]", "word": "split"}, {"info": "", "menu": "[scala]", "kind": "(x$1: String, x$2: Int) => Array[String]", "word": "split"}, {"info": "", "menu": "[scala]", "kind": "() => String", "word": "intern"}, {"info": "", "menu": "[scala]", "kind": "(x$1: String) => String", "word": "concat"}, {"info": "", "menu": "[scala]", "kind": "(x$1: Any) => Boolean", "word": "equals"}, {"info": "", "menu": "[scala]", "kind": "(x$1: Int) => Char", "word": "charAt"}, {"info": "", "menu": "[scala]", "kind": "() => Int", "word": "length"}, {"info": "", "menu": "[scala]", "kind": "(x$1: CharSequence, x$2: CharSequence) => String", "word": "replace"}, {"info": "", "menu": "[scala]", "kind": "(x$1: Char, x$2: Char) => String", "word": "replace"}, {"info": "", "menu": "[scala]", "kind": "(x$1: String) => Boolean", "word": "matches"}, {"info": "", "menu": "[scala]", "kind": "(x$1: String, x$2: Int) => Int", "word": "indexOf"}, {"info": "", "menu": "[scala]", "kind": "(x$1: String) => Int", "word": "indexOf"}, {"info": "", "menu": "[scala]", "kind": "(x$1: Int, x$2: Int) => Int", "word": "indexOf"}, {"info": "", "menu": "[scala]", "kind": "(x$1: Int) => Int", "word": "indexOf"}, {"info": "", "menu": "[scala]", "kind": "() => Boolean", "word": "isEmpty"}, {"info": "", "menu": "[scala]", "kind": "() => String", "word": "toString"}, {"info": "", "menu": "[scala]", "kind": "(x$1: CharSequence) => Boolean", "word": "contains"}, {"info": "", "menu": "[scala]", "kind": "() => Int", "word": "hashCode"}, {"info": "", "menu": "[scala]", "kind": "(x$1: String) => Boolean", "word": "endsWith"}, {"info": "", "menu": "[scala]", "kind": "() => Array[Byte]", "word": "getBytes"}, {"info": "", "menu": "[scala]", "kind": "(x$1: Charset) => Array[Byte]", "word": "getBytes"}, {"info": "", "menu": "[scala]", "kind": "(x$1: String) => Array[Byte]", "word": "getBytes"}, {"info": "", "menu": "[scala]", "kind": "(x$1: Int, x$2: Int, x$3: Array[Byte], x$4: Int) => Unit", "word": "getBytes"}, {"info": "", "menu": "[scala]", "kind": "(x$1: Int, x$2: Int, x$3: Array[Char], x$4: Int) => Unit", "word": "getChars"}, {"info": "", "menu": "[scala]", "kind": "(x$1: Int, x$2: Int) => String", "word": "substring"}, {"info": "", "menu": "[scala]", "kind": "(x$1: Int) => String", "word": "substring"}, {"info": "", "menu": "[scala]", "kind": "(x$1: String) => Int", "word": "compareTo"}, {"info": "", "menu": "[scala]", "kind": "(x$1: String, x$2: String) => String", "word": "replaceAll"}, {"info": "", "menu": "[scala]", "kind": "(x$1: String) => Boolean", "word": "startsWith"}, {"info": "", "menu": "[scala]", "kind": "(x$1: String, x$2: Int) => Boolean", "word": "startsWith"}, {"info": "", "menu": "[scala]", "kind": "() => Array[Char]", "word": "toCharArray"}, {"info": "", "menu": "[scala]", "kind": "() => String", "word": "toUpperCase"}, {"info": "", "menu": "[scala]", "kind": "(x$1: Locale) => String", "word": "toUpperCase"}, {"info": "", "menu": "[scala]", "kind": "() => String", "word": "toLowerCase"}, {"info": "", "menu": "[scala]", "kind": "(x$1: Locale) => String", "word": "toLowerCase"}, {"info": "", "menu": "[scala]", "kind": "(x$1: Int, x$2: Int) => CharSequence", "word": "subSequence"}, {"info": "", "menu": "[scala]", "kind": "(x$1: Int, x$2: Int) => Int", "word": "lastIndexOf"}, {"info": "", "menu": "[scala]", "kind": "(x$1: Int) => Int", "word": "lastIndexOf"}, {"info": "", "menu": "[scala]", "kind": "(x$1: String, x$2: Int) => Int", "word": "lastIndexOf"}, {"info": "", "menu": "[scala]", "kind": "(x$1: String) => Int", "word": "lastIndexOf"}, {"info": "", "menu": "[scala]", "kind": "(x$1: Int) => Int", "word": "codePointAt"}, {"info": "", "menu": "[scala]", "kind": "(x$1: String, x$2: String) => String", "word": "replaceFirst"}, {"info": "", "menu": "[scala]", "kind": "(x$1: Boolean, x$2: Int, x$3: String, x$4: Int, x$5: Int) => Boolean", "word": "regionMatches"}, {"info": "", "menu": "[scala]", "kind": "(x$1: Int, x$2: String, x$3: Int, x$4: Int) => Boolean", "word": "regionMatches"}, {"info": "", "menu": "[scala]", "kind": "(x$1: CharSequence) => Boolean", "word": "contentEquals"}, {"info": "", "menu": "[scala]", "kind": "(x$1: StringBuffer) => Boolean", "word": "contentEquals"}, {"info": "", "menu": "[scala]", "kind": "(x$1: Int, x$2: Int) => Int", "word": "codePointCount"}, {"info": "", "menu": "[scala]", "kind": "(x$1: Int) => Int", "word": "codePointBefore"}, {"info": "", "menu": "[scala]", "kind": "(x$1: String) => Boolean", "word": "equalsIgnoreCase"}, {"info": "", "menu": "[scala]", "kind": "(x$1: Int, x$2: Int) => Int", "word": "offsetByCodePoints"}, {"info": "", "menu": "[scala]", "kind": "(x$1: String) => Int", "word": "compareToIgnoreCase"}, {"info": "", "menu": "[scala]", "kind": "(other: String) => String", "word": "+"}, {"info": "", "menu": "[scala]", "kind": "(groupNames: <repeated>[String]) => Regex", "word": "r"}, {"info": "", "menu": "[scala]", "kind": "Regex", "word": "r"}, {"info": "", "menu": "[scala]", "kind": "(n: Int) => String", "word": "*"}, {"info": "", "menu": "[scala]", "kind": "(that: String) => Boolean", "word": ">"}, {"info": "", "menu": "[scala]", "kind": "(that: String) => Boolean", "word": "<"}, {"info": "", "menu": "[scala]", "kind": "(y: B) => Tuple2[String, B]", "word": "\u2192"}, {"info": "", "menu": "[scala]", "kind": "(that: String) => Boolean", "word": ">="}, {"info": "", "menu": "[scala]", "kind": "(that: String) => Boolean", "word": "<="}, {"info": "", "menu": "[scala]", "kind": "(elem: B) => That", "word": ":+"}, {"info": "", "menu": "[scala]", "kind": "(elem: B) => That", "word": ":+"}, {"info": "", "menu": "[scala]", "kind": "(elem: B) => That", "word": "+:"}, {"info": "", "menu": "[scala]", "kind": "(elem: B) => That", "word": "+:"}, {"info": "", "menu": "[scala]", "kind": "(cbf: CanBuildFrom[Nothing, Char, Col[Char]]) => Col[Char]", "word": "to"}, {"info": "", "menu": "[scala]", "kind": "(that: GenTraversableOnce[B]) => That", "word": "++"}, {"info": "", "menu": "[scala]", "kind": "(that: GenTraversableOnce[B]) => That", "word": "++"}, {"info": "", "menu": "[scala]", "kind": "(z: B) => B", "word": ":\\"}, {"info": "", "menu": "[scala]", "kind": "(z: B) => B", "word": "/:"}, {"info": "", "menu": "[scala]", "kind": "(y: B) => Tuple2[String, B]", "word": "->"}, {"info": "", "menu": "[scala]", "kind": "WrappedString", "word": "seq"}, {"info": "", "menu": "[scala]", "kind": "(that: GenIterable[B]) => That", "word": "zip"}, {"info": "", "menu": "[scala]", "kind": "(that: GenIterable[B]) => That", "word": "zip"}, {"info": "", "menu": "[scala]", "kind": "(f: Function1[Char, B]) => That", "word": "map"}, {"info": "", "menu": "[scala]", "kind": "(f: Function1[Char, B]) => That", "word": "map"}, {"info": "", "menu": "[scala]", "kind": "(that: Traversable[B]) => That", "word": "++:"}, {"info": "", "menu": "[scala]", "kind": "(that: TraversableOnce[B]) => That", "word": "++:"}, {"info": "", "menu": "[scala]", "kind": "(that: Traversable[B]) => That", "word": "++:"}, {"info": "", "menu": "[scala]", "kind": "(that: TraversableOnce[B]) => That", "word": "++:"}, {"info": "", "menu": "[scala]", "kind": "ParSeq", "word": "par"}, {"info": "", "menu": "[scala]", "kind": "(cmp: Ordering[B]) => Char", "word": "max"}, {"info": "", "menu": "[scala]", "kind": "(cmp: Ordering[B]) => Char", "word": "min"}, {"info": "", "menu": "[scala]", "kind": "(num: Numeric[B]) => B", "word": "sum"}, {"info": "", "menu": "[scala]", "kind": "(p: Function1[Char, Boolean]) => Tuple2[String, String]", "word": "span"}, {"info": "", "menu": "[scala]", "kind": "(n: Int) => String", "word": "drop"}, {"info": "", "menu": "[scala]", "kind": "(n: Int) => String", "word": "take"}, {"info": "", "menu": "[scala]", "kind": "String", "word": "init"}, {"info": "", "menu": "[scala]", "kind": "Char", "word": "last"}, {"info": "", "menu": "[scala]", "kind": "String", "word": "tail"}, {"info": "", "menu": "[scala]", "kind": "Char", "word": "head"}, {"info": "", "menu": "[scala]", "kind": "(p: Function1[Char, Boolean]) => Option[Char]", "word": "find"}, {"info": "", "menu": "[scala]", "kind": "(from: Int, until: Int) => SeqView[Char, String]", "word": "view"}, {"info": "", "menu": "[scala]", "kind": "<refinement>", "word": "view"}, {"info": "", "menu": "[scala]", "kind": "(that: GenSeq[B]) => String", "word": "diff"}, {"info": "", "menu": "[scala]", "kind": "Int", "word": "size"}, {"info": "", "menu": "[scala]", "kind": "(z: B) => That", "word": "scan"}, {"info": "", "menu": "[scala]", "kind": "(z: B) => That", "word": "scan"}, {"info": "", "menu": "[scala]", "kind": "(z: A1) => A1", "word": "fold"}]}

```

**Comment by jvican** [comment]
Omnicompletion should accept that and the problem is probably in the details.
  
Here you can find all the documentation for omnicompletion, perhaps I miss something.
http://vimdoc.sourceforge.net/htmldoc/insert.html#complete-functions

**Comment by ChrisCoffey** [comment]
#172 resolves this.

**Comment by ChrisCoffey** [comment]
@jvican This is closed by #171
